src/sentry/api/exceptions.py

```diff
@@ -71,6 +71,15 @@ def __init__(self, user):
         super().__init__(username=user.username)
 
 
+class EmailVerificationRequired(SentryAPIException):
+    status_code = status.HTTP_401_UNAUTHORIZED
+    code = "email-verification-required"
+    message = "Email verification required."
+
+    def __init__(self, user):
+        super().__init__(username=user.username)
```

Comment:
Does it make sense to log email too? There are situations where `user.username` is a random string. 

If it a good idea, we can amend `SudoRequired` too.

Reply:
i don't think its a huge deal, as for further debugging the username can be looked up.